bpf/tests/pktgen.h

```diff
@@ -408,6 +422,46 @@ struct ipv6hdr *pktgen__push_default_ipv6hdr(struct pktgen *builder)
 	return hdr;
 }
 
+/* Push an empty ARP header onto the packet */
+static __always_inline
+__attribute__((warn_unused_result))
+struct arphdreth *pktgen__push_arphdr_ethernet(struct pktgen *builder)
+{
+	struct __ctx_buff *ctx = builder->ctx;
+	struct arphdreth *layer;
+	int layer_idx;
+
+	/* Request additional tailroom, and check that we got it. */
+	ctx_adjust_troom(ctx, builder->cur_off + sizeof(struct arphdreth) - ctx_full_len(ctx));
+	if (ctx_data(ctx) + builder->cur_off + sizeof(struct arphdreth) > ctx_data_end(ctx))
+		return 0;
+
+	/* Check that any value within the struct will not exceed a u16 which
+	 * is the max allowed offset within a packet from ctx->data.
+	 */
+	if (builder->cur_off >= MAX_PACKET_OFF - sizeof(struct arphdreth))
+		return 0;
+
+	layer = ctx_data(ctx) + builder->cur_off;
```

Comment:
I expect this to fail in the CI now that we're using inline assembly to implement `ctx_data()` and `ctx_data_end()`. If that's the case, please make sure you call these two functions only once (`data = ctx_data(ctx)` then use `data` as much as you want), as the compiler can no longer optimise unnecessary calls out. (You still need to call them again after a helper call that invalidates the call though, but that's not the case here as I read it.)

Reply:
This seems to work, every variable on this line is 64bit so no 32-bit verifier bullshit. Unit tests seem happy in CI as is